cobbler/actions/buildiso/netboot.py

```diff
@@ -328,18 +330,22 @@ def _get_tcp_ip_config(self):
             if self.data.get("name_servers") != "":
                 self.system_dns = self.data["name_servers"]
 
-    def generate_system(self, dist, system, exclude_dns: bool) -> str:
+    def generate_system(
+        self, dist, system, exclude_dns: bool, protocol: str = "http"
+    ) -> str:
         """
-        Generate the append line for a netbooting system.
+        Generate the append-line for a net-booting system.
+
         :param dist: The distribution associated with the system.
         :param system: The system itself
         :param exclude_dns: Whether to include the DNS config or not.
+        :param protocol: The protocol that is used to read the autoyast file from the server
         """
         self.dist = dist
 
         self.append_line = f"  APPEND initrd={self.distro_name}.img"
         if self.dist.breed == "suse":
-            self._generate_append_suse()
+            self._generate_append_suse(protocol=protocol)
         elif self.dist.breed == "redhat":
             self._generate_append_redhat()
```

Comment:
Is this change for only one implementation of `_generate_append()` intended? Would it make sense to keep the implementations aligned? Or are they to different?

Reply:
The implementations are different as the different installers (kickstart, autoyast and preseed) should actually not distro names but installer names. Thus it doesn't make sense to align the implementations as the capabilites and requirements of the different installers are not similar.